### PR TITLE
refactor(client): use skipToken for RTK queries

### LINKED ISSUE #2998: Refactor: try to use RTK Query's skip token
See: https://redux-toolkit.js.org/rtk-query/usage-with-typescript#skipping-queries-with-typescript-using-skiptoken

This would provide better typing, compare:
```ts
const { data } = useMyQuery({ branch: branch ?? "" }, { skip: !branch });
```
 with
```ts
const { data } = useMyQuery(!!branch ? { branch } : skipToken);
```

### PR DESCRIPTION
**Pull request by leafty**
Fixes #2998.

Use `skipToken` when RTK queries need to be skipped, which provides better typing as we avoid using `?? 0`, `?? ""`, `?? []`, etc.

/deploy

### CODE REVIEW

File: client/src/components/ssh/ssh.tsx [hunk 1]
```diff
@@ -84,12 +85,13 @@ function SshModal() {
     undefined // do not use the override for getting migration status
   );
   const coreSupport = projectCoreApi.useGetMigrationStatusQuery(
-    {
-      apiVersion: migrationStatusApiVersion,
-      gitUrl,
-      branch: displayModal.branch,
-    },
-    { skip: !gitUrl || !displayModal.branch }
+    !!gitUrl && !!displayModal.branch
```
Comment (leafty) [inline client/src/components/ssh/ssh.tsx:88]:
```suggestion
    gitUrl && displayModal.branch
```

File: client/src/features/project/components/ProjectEntityHeader.tsx [hunk 2]
```diff
@@ -35,13 +37,14 @@ export function ProjectEntityHeader(props: ProjectEntityHeaderProps) {
   const { defaultBranch, devAccess, fullPath, gitUrl, projectId, visibility } =
     props;
 
-  const projectIndexingStatus = useGetProjectIndexingStatusQuery(projectId, {
-    skip: !fullPath || !projectId,
-  });
+  const projectIndexingStatus = useGetProjectIndexingStatusQuery(
+    !!fullPath && !!projectId ? projectId : skipToken
```
Comment (leafty) [inline client/src/features/project/components/ProjectEntityHeader.tsx:41]:
```suggestion
    fullPath && projectId ? projectId : skipToken
```

File: client/src/features/project/components/ProjectEntityHeader.tsx [hunk 3]
```diff
@@ -35,13 +37,14 @@ export function ProjectEntityHeader(props: ProjectEntityHeaderProps) {
...
+  );
 
   const projectMetadataQuery = useProjectMetadataQuery(
-    { projectPath: fullPath, projectId },
-    { skip: !fullPath || !projectId || !projectIndexingStatus.data?.activated }
+    !!fullPath && !!projectId && !!projectIndexingStatus.data?.activated
```
Comment (leafty) [inline client/src/features/project/components/ProjectEntityHeader.tsx:45]:
```suggestion
    fullPath && projectId && projectIndexingStatus.data?.activated
```

File: client/src/features/project/components/ProjectPageTitle.tsx [hunk 4]
```diff
@@ -35,21 +36,19 @@ function ProjectPageTitle({
   projectPathWithNamespace,
   projectTitle,
 }: ProjectJsonLdProps) {
-  const projectIndexingStatus = useGetProjectIndexingStatusQuery(projectId, {
-    skip: !projectPathWithNamespace || !projectId,
-  });
+  const projectIndexingStatus = useGetProjectIndexingStatusQuery(
+    !!projectPathWithNamespace && !!projectId ? projectId : skipToken
```
Comment (leafty) [inline client/src/features/project/components/ProjectPageTitle.tsx:40]:
```suggestion
    projectPathWithNamespace && projectId ? projectId : skipToken
```

File: client/src/features/project/components/ProjectSettingAvatar.tsx [hunk 5]
```diff
@@ -71,17 +73,13 @@ export function ProjectSettingsAvatar({
 }: ProjectSettingsAvatarProps) {
   const [avatar, setAvatar] = useState<ImageValue>();
   const [succeeded, setSucceeded] = useState<boolean | undefined>(undefined);
-  const projectIndexingStatus = useGetProjectIndexingStatusQuery(projectId, {
-    skip: !projectFullPath || !projectId,
-  });
+  const projectIndexingStatus = useGetProjectIndexingStatusQuery(
+    !!projectFullPath && !!projectId ? projectId : skipToken
```
Comment (leafty) [inline client/src/features/project/components/ProjectSettingAvatar.tsx:77]:
```suggestion
    projectFullPath && projectId ? projectId : skipToken
```

File: client/src/features/project/components/ProjectSettingAvatar.tsx [hunk 6]
```diff
@@ -71,17 +73,13 @@ export function ProjectSettingsAvatar({
...
+  );
   const projectMetadata = useProjectMetadataQuery(
-    { projectPath: projectFullPath, projectId },
-    {
-      skip:
-        !projectFullPath ||
-        !projectId ||
-        !projectIndexingStatus.data?.activated,
-    }
+    !!projectFullPath && !!projectId && !projectIndexingStatus.data?.activated
```
Comment (leafty) [inline client/src/features/project/components/ProjectSettingAvatar.tsx:80]:
```suggestion
    projectFullPath && projectId && projectIndexingStatus.data?.activated
```

File: client/src/features/project/components/ProjectSettingsDescription.tsx [hunk 7]
```diff
@@ -50,17 +52,13 @@ export function ProjectSettingsDescription({
   const [succeeded, setSucceeded] = React.useState<boolean | undefined>(
     undefined
   );
-  const projectIndexingStatus = useGetProjectIndexingStatusQuery(projectId, {
-    skip: !projectFullPath || !projectId,
-  });
+  const projectIndexingStatus = useGetProjectIndexingStatusQuery(
+    !!projectFullPath && !!projectId ? projectId : skipToken
```
Comment (leafty) [inline client/src/features/project/components/ProjectSettingsDescription.tsx:56]:
```suggestion
    projectFullPath && projectId ? projectId : skipToken
```

File: client/src/features/project/components/ProjectSettingsDescription.tsx [hunk 8]
```diff
@@ -50,17 +52,13 @@ export function ProjectSettingsDescription({
...
+  );
   const projectMetadata = useProjectMetadataQuery(
-    { projectPath: projectFullPath, projectId },
-    {
-      skip:
-        !projectFullPath ||
-        !projectId ||
-        !projectIndexingStatus.data?.activated,
-    }
+    !!projectFullPath && !!projectId && !!projectIndexingStatus.data?.activated
```
Comment (leafty) [inline client/src/features/project/components/ProjectSettingsDescription.tsx:59]:
```suggestion
    projectFullPath && projectId && projectIndexingStatus.data?.activated
```

File: client/src/features/session/components/SessionRowCommitInfo.tsx [hunk 9]
```diff
@@ -62,8 +64,7 @@ function SessionRowCommitInfoDetails({
     isLoading,
     isError,
   } = useGetRepositoryCommitQuery(
-    { commitSha, projectId },
-    { skip: !commitSha || !projectId }
+    !!commitSha && !!projectId ? { commitSha, projectId } : skipToken
```
Comment (leafty) [inline client/src/features/session/components/SessionRowCommitInfo.tsx:67]:
```suggestion
    commitSha && projectId ? { commitSha, projectId } : skipToken
```

File: client/src/features/session/components/options/AutostartSessionOptions.tsx [hunk 10]
```diff
@@ -115,18 +116,20 @@ function useAutostartSessionOptions(): void {
...
+        : skipToken
     );
   const { data: commits, isFetching: commitsIsFetching } =
     useGetAllRepositoryCommitsQuery(
-      {
-        branch: currentBranch,
-        projectId: `${gitLabProjectId ?? 0}`,
-      },
-      { skip: !gitLabProjectId || !currentBranch }
+      !!gitLabProjectId && !!currentBranch
```
Comment (leafty) [inline client/src/features/session/components/options/AutostartSessionOptions.tsx:127]:
```suggestion
      gitLabProjectId && currentBranch
```

File: client/src/features/session/components/options/AutostartSessionOptions.tsx [hunk 11]
```diff
@@ -143,40 +146,37 @@ function useAutostartSessionOptions(): void {
...
-      metadataVersion,
-      projectRepositoryUrl,
-      branch: currentBranch,
-      commit,
-    },
-    {
-      skip:
-        !backendAvailable || !coreSupportComputed || !currentBranch || !commit,
-    }
+    !!backendAvailable && coreSupportComputed && !!currentBranch && !!commit
```
Comment (leafty) [inline client/src/features/session/components/options/AutostartSessionOptions.tsx:149]:
```suggestion
    backendAvailable && coreSupportComputed && currentBranch && commit
```

File: client/src/features/session/components/options/AutostartSessionOptions.tsx [hunk 12]
```diff
@@ -143,40 +146,37 @@ function useAutostartSessionOptions(): void {
...
-      { project_id: `${gitLabProjectId}` },
-      {
-        skip:
-          !gitLabProjectId ||
-          !notebooksVersion ||
-          !notebooksVersion.cloudStorageEnabled,
-      }
+      !!gitLabProjectId &&
+        !!notebooksVersion &&
+        !!notebooksVersion.cloudStorageEnabled
```
Comment (leafty) [inline client/src/features/session/components/options/AutostartSessionOptions.tsx:177]:
```suggestion
      gitLabProjectId &&
        notebooksVersion &&
        notebooksVersion.cloudStorageEnabled
```

File: client/src/features/session/components/options/SessionCommitOption.tsx [hunk 13]
```diff
@@ -65,11 +66,12 @@ export default function SessionCommitOption() {
     isFetching,
     requestId,
   } = useGetRepositoryCommitsQuery(
-    {
-      branch: currentBranch,
-      projectId: `${gitLabProjectId}`,
-    },
-    { skip: !gitLabProjectId || !currentBranch }
+    !!gitLabProjectId && !!currentBranch
```
Comment (leafty) [inline client/src/features/session/components/options/SessionCommitOption.tsx:69]:
```suggestion
    gitLabProjectId && currentBranch
```

File: client/src/features/session/components/options/SessionStorageOption.tsx [hunk 14]
```diff
@@ -58,16 +59,15 @@ export const SessionStorageOption = () => {
...
-      apiVersion,
-      metadataVersion,
-      projectRepositoryUrl,
-      branch: currentBranch,
-      commit,
-    },
-    {
-      skip: !coreSupportComputed || !currentBranch || !commit,
-    }
+    !!coreSupportComputed && !!currentBranch && !!commit
```
Comment (leafty) [inline client/src/features/session/components/options/SessionStorageOption.tsx:62]:
```suggestion
    coreSupportComputed && currentBranch && commit
```

File: client/src/features/session/components/options/StartNotebookServerOptions.tsx [hunk 15]
```diff
@@ -389,16 +380,15 @@ const AutoFetchLfsOption = () => {
...
-      apiVersion,
-      metadataVersion,
-      projectRepositoryUrl,
-      branch: currentBranch,
-      commit,
-    },
-    {
-      skip: !coreSupportComputed || !currentBranch || !commit,
-    }
+    !!coreSupportComputed && !!currentBranch && !!commit
```
Comment (leafty) [inline client/src/features/session/components/options/StartNotebookServerOptions.tsx:383]:
```suggestion
    coreSupportComputed && currentBranch && commit
```

File: client/src/project/shared/ProjectKeywords.tsx [hunk 16]
```diff
@@ -75,17 +76,13 @@ function ProjectKeywordsInput({
   };
 
   const [keywords, setKeywords] = useState<string>("");
-  const projectIndexingStatus = useGetProjectIndexingStatusQuery(projectId, {
-    skip: !projectFullPath || !projectId,
-  });
+  const projectIndexingStatus = useGetProjectIndexingStatusQuery(
+    !!projectFullPath && !!projectId ? projectId : skipToken
```
Comment (leafty) [inline client/src/project/shared/ProjectKeywords.tsx:80]:
```suggestion
    projectFullPath && projectId ? projectId : skipToken
```

File: client/src/project/shared/ProjectKeywords.tsx [hunk 17]
```diff
@@ -75,17 +76,13 @@ function ProjectKeywordsInput({
...
+  );
   const projectMetadata = useProjectMetadataQuery(
-    { projectPath: projectFullPath, projectId },
-    {
-      skip:
-        !projectFullPath ||
-        !projectId ||
-        !projectIndexingStatus.data?.activated,
-    }
+    !!projectFullPath && !!projectId && !!projectIndexingStatus.data?.activated
```
Comment (leafty) [inline client/src/project/shared/ProjectKeywords.tsx:83]:
```suggestion
    projectFullPath && projectId && projectIndexingStatus.data?.activated
```

File: client/src/utils/customHooks/useGetRecentlyVisitedProjects.ts [hunk 18]
```diff
@@ -16,15 +16,17 @@
...
+
 import { useGetRecentlyVisitedProjectsQuery } from "../../features/projects/projects.api";
 import { Session } from "../helpers/SessionFunctions";
 
 interface UseGetRecentlyVisitedProjectsArgs {
   projectsCount: number;
   currentSessions: Session[];
   pinnedProjectSlugs: string[];
-  skip?: boolean;
+  // skip?: boolean;
```
Comment (leafty) [inline client/src/utils/customHooks/useGetRecentlyVisitedProjects.ts:29]:
```suggestion
```

File: client/src/workflows/Workflows.container.tsx [hunk 19]
```diff
@@ -129,17 +137,17 @@ function WorkflowsList({
...
-    {
-      apiVersion,
-      gitUrl: repositoryUrl,
-      metadataVersion,
-      workflowId: selected,
-      reference,
-      fullPath,
-    },
-    { skip: skipDetails }
+    !skipList && !!selected
```
Comment (leafty) [inline client/src/workflows/Workflows.container.tsx:141]:
```suggestion
    !skipList && selected
```

**Review by leafty — COMMENTED** [review]
(using the GH interface for adjusting the PR)

**Review by lorenzo-cavazzi — APPROVED** [review]

**Pull request merged**